src/include/daos_srv/bio.h

```diff
@@ -86,6 +86,26 @@ struct bio_blob_hdr {
 	uuid_t		bbh_pool;
 };
 
+/*
+ * Current device state (health statistics). Periodically updated in
+ * bio_bs_monitor(). Used to determine faulty device status.
+ */
+struct bio_device_health_state {
+	uint64_t	 bhs_timestamp;
+	int		 bhs_bio_err;
+	/* Health log page information */
+	uint64_t	*bhs_media_errors; /* supports 128-bit values */
+	uint16_t	 bhs_temperature; /* in Kelvin */
+	/* Critical warnings */
+	uint8_t		 bhs_temp_warning	: 1;
+	uint8_t		 bhs_avail_spare_warning	: 1;
+	uint8_t		 bhs_dev_reliabilty_warning : 1;
+	uint8_t		 bhs_read_only_warning	: 1;
+	uint8_t		 bhs_volatile_mem_warning: 1; /*volatile memory backup*/
```

Comment:
How will these bits are used? Could you explain it more?

Reply:
like if the bit is set, meaning there was a critical warning triggered on the NVMe SSD, then that would be an important consideration/criteria for faulty device detection I would think. I included all the critical warning currently logged on device, but we could start with just like a temp warning and dev reliability warning if that seems more appropriate? or are you thinking remove all bit warnings in general?

Reply:
I see, I just didn't know they are representing warnings triggered on SSD. I think they are all important for faulty criteria.